cpp/src/phonenumbers/phonenumbermatcher.cc

```diff
@@ -626,6 +641,10 @@ bool PhoneNumberMatcher::ExtractMatch(const string& candidate, int offset,
 }
 
 bool PhoneNumberMatcher::HasNext() {
+  if (!isInputInCleanUTF8_) {
```

Comment:
This is where I would put a comment explaining *why* we early exit if it is not valid utf8 :)

Reply:
Done. Thanks Lara :)